codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -469,8 +472,108 @@ class ServerServiceGeneratorV2(
+                }
+            }
+
+        val hasErrors = service.operations.any { model.expectShape(it).asOperationShape().get().errors.isNotEmpty() }
+
         writer.rustTemplate(
             """
+            /// A fast and customizable Rust implementation of the $serviceName Smithy service.
+            ///
+            /// The primary export is [`$serviceName`]: it satisfies the [`Service<http::Request, Response = http::Response>`]
+            /// trait and therefore can be handed to [Hyper server] using [`$serviceName::into_make_service`] or used in Lambda using [`#{SmithyHttpServer}::routing::LambdaHandler`].
+            /// The [`crate::${InputsModule.name}`], ${if (!hasErrors) "and " else ""}[`crate::${OutputsModule.name}`], ${if (hasErrors) "and [`crate::${ErrorsModule.name}`]" else "" }
+            /// modules provide the types used in each operation.
+            ///
+            /// The primary export is [`$serviceName`]: it is the
+            /// [`$builderName`] is used to set your business logic to implement your [operations],
+            /// customize your [operations]'s behaviors by applying middleware,
+            /// and build your service.
+            ///
+            /// [`$builderName`] contains the [operations] modeled in your Smithy service.
+            /// You must set an implementation for all operations with the correct signature,
+            /// or your service will fail to be constructed at runtime and panic.
+            /// For each of the [operations] modeled in
+            /// your Smithy service, you need to provide an implementation in the
+            /// form of an async function that takes in the
+            /// operation's input as their first parameter, and returns the
+            /// operation's output. If your operation is fallible (i.e. it
+            /// contains the `errors` member in your Smithy model), the function
+            /// implementing the operation has to be fallible (i.e. return a [`Result`]).
+            /// The possible forms for your async functions are:
+            /// ```rust,no_run
+            /// async fn handler_fallible(input: Input, extensions: #{SmithyHttpServer}::Extension<T>) -> Result<Output, Error>
+            /// async fn handler_infallible(input: Input, extensions: #{SmithyHttpServer}::Extension<T>) -> Output
+            /// ```
+            /// Both can take up to 8 extensions, or none:
+            /// ```rust,no_run
+            /// async fn handler_with_no_extensions(input: Input) -> ...;
+            /// async fn handler_with_one_extension(input: Input, ext: #{SmithyHttpServer}::Extension<T>) -> ...
+            /// async fn handler_with_two_extensions(input: Input, ext0: #{SmithyHttpServer}::Extension<T>, ext1: #{SmithyHttpServer}::Extension<T>) -> ...
+            /// ...
+            /// ```
+            /// For a full list of the possible extensions, see: [`#{SmithyHttpServer}::request`]. Any `T: Send + Sync + 'static` is also allowed.
```

Comment:
> Any `T: Send + Sync + 'static` is also allowed

Miscommunication by me here, any `Extension<T>` implements `FromParts` when `T: Send + Sync + 'static`, the `T` itself doesn't.